pkg/apis/v1/ec2nodeclass_status.go

```diff
@@ -54,6 +55,9 @@ type AMI struct {
 	// ID of the AMI
 	// +required
 	ID string `json:"id"`
+	// Deprecation status of the AMI
+	// +optional
+	Deprecated *bool `json:"deprecated,omitempty"`
```

Comment:
```suggestion
	Deprecated bool `json:"deprecated,omitempty"`
```

Reply:
Using a pointer here to set the `Deprecated` field to nil, so we only show this field on the AMI status if the AMI is deprecated


```
status:
  amis:
  - id: ami-01234567890654321
    name: amazon-eks-node-1.29
    deprecated: true
    requirements:
    - key: kubernetes.io/arch
      operator: In
      values:
      - amd64
  # Non deprecated AMI will not show the field deprecated #
  - id: ami-01234567890123456
    name: amazon-eks-arm64-node-1.29
    requirements:
    - key: kubernetes.io/arch
      operator: In
      values:
      - arm64

Reply:
You don't need a pointer to do that from my understand. `omitempty` will make sure it's nil, no?

Reply:
False is treated as empty so omitempty will drop it automatically, @engedaam is correct that you can just use a non-pointer type here: https://go.dev/play/p/_3p3MCAFhN5

Reply:
Thank you folks :) updated in the latest commit